### PR TITLE
update rt.conf and compile.sh to provide cmake options and avoid a translation layer

### PR DESCRIPTION
**Pull request by aerorahul**
# PR Checklist

- [ ] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. 
- [ ] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR
- [ ] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. 

## Description

`compile.sh` reads in the `MAKE_OPT` which are the deprecated GNU-style options for compilations from `rt.conf`.
This PR:
- replaces GNU-style compile options from `rt.conf` in `cmake` style.
- updates `compile.sh` to pass through `MAKE_OPT` as is to `build.sh` which calls `cmake`.
- removes confusion as to what is an input to `compile.sh` and `build.sh`.  E.g. `SUITES` is (was) input to `MAKE_OPT` and `compile.sh` would transform that to `-DCCPP_SUITES` for use in `build.sh`.  Hereafter, we use `-DCCPP_SUITES` in `rt.conf`  Similar changes have been made to `32BIT`, `DEBUG`, `REPRO`, `MULTI_GASES`, etc.

### Issue(s) addressed
Part of #416 
More work might be needed based on the comments to this PR.

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [ ] hera.intel
- [ ] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel 
- [ ] cheyenne.gnu
- [ ] gaea.intel 
- [ ] jet.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3

## Dependencies
None.

### CODE REVIEW

**Comment by DusanJovic-NOAA** [comment]
`rt.sh` must be updated at two places where `MAKE_OPT` is used:

`if [[ ${MAKE_OPT^^} =~ "REPRO=Y" ]]; then`

and:

`if [[ ${MAKE_OPT^^} =~ "WW3=Y" ]]; then`

**Comment by aerorahul** [comment]
This is going to cause conflicts everytime develop is updated, so I am going to wait till it is this PR's turn to resolve all conflicts.
I will fix the issue in [this](https://github.com/ufs-community/ufs-weather-model/pull/615#issuecomment-853111692) comment needing an update in `rt.sh`.

**Comment by aerorahul** [comment]
> `rt.sh` must be updated at two places where `MAKE_OPT` is used:
> 
> `if [[ ${MAKE_OPT^^} =~ "REPRO=Y" ]]; then`
> 
> and:
> 
> `if [[ ${MAKE_OPT^^} =~ "WW3=Y" ]]; then`

There is no option with `WW3=Y|N` in `MAKE_OPT` in any of the `conf` files.

**Comment by DusanJovic-NOAA** [comment]
> > `rt.sh` must be updated at two places where `MAKE_OPT` is used:
> > `if [[ ${MAKE_OPT^^} =~ "REPRO=Y" ]]; then`
> > and:
> > `if [[ ${MAKE_OPT^^} =~ "WW3=Y" ]]; then`
> 
> There is no option with `WW3=Y|N` in `MAKE_OPT` in any of the `conf` files.

I see. Then this second if test must be updated to look for APP that includes WW3 (ATMW, S2SW, etc).

**Comment by aerorahul** [comment]
> > > `rt.sh` must be updated at two places where `MAKE_OPT` is used:
> > > `if [[ ${MAKE_OPT^^} =~ "REPRO=Y" ]]; then`
> > > and:
> > > `if [[ ${MAKE_OPT^^} =~ "WW3=Y" ]]; then`
> > 
> > 
> > There is no option with `WW3=Y|N` in `MAKE_OPT` in any of the `conf` files.
> 
> I see. Then this second if test must be updated to look for APP that includes WW3 (ATMW, S2SW, etc).

done in 0905841.  Please take a look again.

**Comment by uturuncoglu** [comment]
@aerorahul i am not sure but is this PR is for solving the issue that I faced before. The build was trying to add multiple APP in the build.

**Comment by aerorahul** [comment]
> @aerorahul i am not sure but is this PR is for solving the issue that I faced before. The build was trying to add multiple APP in the build.

@uturuncoglu Yes I believe so.  In this PR, the contents of `MAKE_OPT` are directly passed to cmake.  `compile.sh` does not _translate_ `MAKE_OPT` into `CMAKE_FLAGS`.   This will resolve the issue you faced where the build was adding multiple APP's to the build.  Please take a look and let me know if the issue persists or creates another one.

**Comment by uturuncoglu** [comment]
@aerorahul Thanks for the clarification. Currently I could not access to Orion and I am trying to setup Cheyenne for testing. I'll let you know when I have a change to test it. Thanks again.

**Comment by uturuncoglu** [comment]
@aerorahul any update about this PR?

**Comment by uturuncoglu** [comment]
@aerorahul @junwang-noaa i am testing this PR along with the https://github.com/ufs-community/ufs-weather-model/pull/611 and I am getting build error like following when I tried to compile DOCN case,

```
+ cmake /work/noaa/gmtb/tufuk/UFS_cdeps APP=HAFS_DOCN SUITES=HAFS_v0_gfdlmp_tedmf_nonsst 32BIT=Y -DMPI=ON -DCMAKE_BUILD_TYPE=Release
CMake Error: The source directory "/work/noaa/gmtb/tufuk/UFS_cdeps/tests/build_fv3_001/32BIT=Y" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
```

any idea. I'll also review the PR in a more detailed way but first I need to test it under HAFS app with data component configurations.

**Comment by uturuncoglu** [comment]
maybe I need to replace `32BIT=Y` with `-D32BIT=ON`

**Comment by uturuncoglu** [comment]
i also need to add -D before CCPP and APP

**Comment by uturuncoglu** [comment]
@aerorahul okay I pass those errors by adding -D and aand make change in 32BIT option. Now, I am getting following error,

```
CMake Error: install(EXPORT "CDEPSExports" ...) includes target "dshr" which requires target "streams" that is not in any export set.
CMake Error: install(EXPORT "CDEPSExports" ...) includes target "datm" which requires target "streams" that is not in any export set.
CMake Error: install(EXPORT "CDEPSExports" ...) includes target "dice" which requires target "streams" that is not in any export set.
CMake Error: install(EXPORT "CDEPSExports" ...) includes target "dlnd" which requires target "streams" that is not in any export set.
CMake Error: install(EXPORT "CDEPSExports" ...) includes target "docn" which requires target "streams" that is not in any export set.
CMake Error: install(EXPORT "CDEPSExports" ...) includes target "drof" which requires target "streams" that is not in any export set.
CMake Error: install(EXPORT "CDEPSExports" ...) includes target "dwav" which requires target "streams" that is not in any export set.
CMake Error in CDEPS-interface/CMakeLists.txt:
  export called with target "dshr" which requires target "streams" that is
  not in any export set.


CMake Error in CDEPS-interface/CMakeLists.txt:
  export called with target "datm" which requires target "streams" that is
  not in any export set.
```

Do I need to use special version of CDEPS? We are using NOAA-EMC CDEPS fork as a base in HAFS app but I have additional modifications for ERA5 data atmosphere in it. So, I am not expecting any issue with the CDEPS version that I am using.

**Comment by aerorahul** [comment]
@uturuncoglu 
I apologize, I have not updated my branch with the latest `develop`.  I will do so now.

**Comment by uturuncoglu** [comment]
@aerorahul okay. no problem. let me know when it is ready and I could test it again.

**Comment by aerorahul** [comment]
@uturuncoglu I think I resolved the conflicts.  I have not tested them yet.

File: tests/rt.sh [hunk 1]
```diff
@@ -632,15 +632,15 @@ EOF
...
+    if [[ ${MAKE_OPT^^} =~ "-DREPRO=ON" ]]; then
       RT_SUFFIX="_repro"
       BL_SUFFIX="_repro"
     else
       RT_SUFFIX=""
       BL_SUFFIX=""
     fi
 
-    if [[ ${MAKE_OPT^^} =~ "WW3=Y" ]]; then
+    if [[ ${MAKE_OPT^^} =~ "-DAPP=ATMW" ]] ||  [[ ${MAKE_OPT^^} =~ "-DAPP=S2SW" ]]; then
```
Comment (uturuncoglu) [inline tests/rt.sh:643]: I think that this is not flexible that I explain in the following issue https://github.com/ufs-community/ufs-weather-model/issues/656. I already completed WW3 coupling through the use of CMEPS under HAFS application and created a set of wave coupled configurations. Listing applications names in this level is not flexible and once someone bring another configuration this file needs to be changed.

Comment (aerorahul) [reply tests/rt.sh:643]: Please make a suggestion/PR and we can review.
All the applications that are present in the UFS-weather-model are tested and supported.  So if someone brings in an "application", that needs to be supported.
The entire business of adding `if-then-else` needs to be revisited.

**Comment by aerorahul** [comment]
#642 absorbed this PR.
Closing.

**Pull request closed without merging**